### DISCUSSION TITLE
next-urql not prefetching subsequent pages via next/link

### DISCUSSION THREAD

**Discussion by andreialecu** [urql-graphql/urql#2116] [Q&A] [answered]
I've been trying to look for documentation on this, but I can't seem to find anything relevant.

It appears that prefetching pages via `next/link`'s automated prefetching does not work when using `useQuery` hooks. Navigation to the target page always executes a client side `graphql` query once the page is loaded.

The `next/link` documentation mentions:

> prefetch - Prefetch the page in the background. Defaults to true. Any <Link /> that is in the viewport (initially or through scroll) will be preloaded.

Is this expected? Is there any way to allow nextjs' default prefetching behavior to work?

**Comment by JoviDeCroock** [comment, marked as answer]
This is not possible until either first-class support for `suspense` is present in Next, you can however consider using `getServerSideProps`/`getStaticProps` which can circumvent this issue at the cost of being a bit more vigilant when building out your application.

**Reply by andreialecu** [reply]
Thank you
